### DISCUSSION TITLE
fix: Replica can't respond to Updater rotation

### DISCUSSION THREAD

**Discussion by anna-carroll** [celo-org/optics-monorepo#604] [Ideas]
there is no `setUpdater` on the Replica contract; if the Home contract changes its Updater, the Replica can't follow suit

note: needs some thought / design around who would report a changed updater

**Comment by luketchang** [comment]
On boot-up, we could have `watcher` check if all replicas it has permissions for have matching updater with `home` referenced in the `ConnectionManager` contract. If not, we could have watcher submit _signed_ a `UpdaterRotation` message to either its permissioned replica directly or to the owning `ConnectionManager` contract, which would forward the message to the replica (makes use of permissioning already built into ConnectionManager). This would put the responsibility of updating replicas on owners/watchers of the replicas, which is more localized.

The other alternative I can think of is going the "governance route" where we use the gov system to administer this change to the replicas. One problem I see with this is that if the updater changes, any linked replicas will not be accepting messages enqueued on the home with the new updater (since they are signed with a new updater the replica is not aware of). And with this problem its not clear how replicas will be able to accept governance messages.
